pkg/images/backend/knhelpers.go

```diff
@@ -0,0 +1,192 @@
+///////////////////////////////////////////////////////////////////////
+// Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+///////////////////////////////////////////////////////////////////////
+
+package backend
+
+import (
+	"encoding/json"
+
+	"github.com/go-openapi/strfmt"
+	knbuild "github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/pkg/errors"
+	dapi "github.com/vmware/dispatch/pkg/api/v1"
+	"github.com/vmware/dispatch/pkg/utils"
+	"github.com/vmware/dispatch/pkg/utils/knaming"
+
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var (
+	defaultDockerfile         = "Dockerfile"
+	defaultSystemPackagesFile = "system-packages.txt"
+	defaultPackagesFile       = "packages.txt"
+)
+
+// FromBaseImage produced Knative BuildTemplate from Dispatch BaseImage
+func FromBaseImage(baseimage *dapi.BaseImage) *knbuild.BuildTemplate {
+	if baseimage == nil {
+		return nil
+	}
+
+	// TODO: BaseImage -> BuildTemplate make this right and configurable
+	return &knbuild.BuildTemplate{
+		ObjectMeta: knaming.ToObjectMeta(baseimage.Meta, *baseimage),
+		Spec: knbuild.BuildTemplateSpec{
+			Parameters: []knbuild.ParameterSpec{
+				knbuild.ParameterSpec{
+					Name:        string("DESTINATION"),
+					Description: string("The destination to push the image"),
+				},
+				knbuild.ParameterSpec{
+					Name:        string("BASE_IMAGE"),
+					Description: string("The base image which this image is built from"),
+				},
+				knbuild.ParameterSpec{
+					Name:        string("DOCKERFILE"),
+					Description: string("Path to the Dockerfile to build"),
+					Default:     &defaultDockerfile,
+				},
+				knbuild.ParameterSpec{
+					Name:        string("SYSTEM_PACKAGES_CONTENT"),
+					Description: string("System packages content that will be written to SYSTEM_PACKAGES_FILE"),
+				},
+				knbuild.ParameterSpec{
+					Name:        string("SYSTEM_PACKAGES_FILE"),
+					Description: string("Path to file with system dependencies"),
+					Default:     &defaultSystemPackagesFile,
+				},
+				knbuild.ParameterSpec{
+					Name:        string("PACKAGES_CONTENT"),
+					Description: string("Runtime packages content that will be written to PACKAGES_FILE"),
+				},
+				knbuild.ParameterSpec{
+					Name:        string("PACKAGES_FILE"),
+					Description: string("Path to file with runtime dependencies"),
+					Default:     &defaultPackagesFile,
+				},
+			},
+			Steps: []corev1.Container{
+				corev1.Container{
+					Name:    string("write-system-package-files"),
+					Image:   string("vmware/photon2:20180424"),
+					Command: []string{"/bin/bash"},
+					Args:    []string{"-c", "echo -n ${SYSTEM_PACKAGES_CONTENT} | base64 -d > /workspace/${SYSTEM_PACKAGES_FILE}"},
+				},
+				corev1.Container{
+					Name:    string("write-package-files"),
+					Image:   string("vmware/photon2:20180424"),
```

Comment:
Any specific reason we need this tag? and not `latest`

Reply:
Will change to latest, forgot to change this while playing with knative. Thanks!